src/impl_riscv_linux.c

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cpu_features_macros.h"
+
+#ifdef CPU_FEATURES_ARCH_RISCV
+#if defined(CPU_FEATURES_OS_LINUX)
+
+#include "cpuinfo_riscv.h"
+
+////////////////////////////////////////////////////////////////////////////////
+// Definitions for introspection.
+////////////////////////////////////////////////////////////////////////////////
+#define INTROSPECTION_TABLE                                      \
+  LINE(RISCV_32, riscv32, "32", RISCV_HWCAP_32, 0)               \
+  LINE(RISCV_64, riscv64, "64", RISCV_HWCAP_64, 0)               \
+  LINE(RISCV_128, riscv128, "128", RISCV_HWCAP_128, 0)           \
+  LINE(RISCV_A, a, "a", RISCV_HWCAP_A, 0)                        \
+  LINE(RISCV_C, c, "c", RISCV_HWCAP_C, 0)                        \
+  LINE(RISCV_D, d, "d", RISCV_HWCAP_D, 0)                        \
+  LINE(RISCV_E, e, "e", RISCV_HWCAP_E, 0)                        \
+  LINE(RISCV_F, f, "f", RISCV_HWCAP_F, 0)                        \
+  LINE(RISCV_I, i, "i", RISCV_HWCAP_I, 0)                        \
+  LINE(RISCV_M, m, "m", RISCV_HWCAP_M, 0)                        \
+  LINE(RISCV_V, v, "v", RISCV_HWCAP_V, 0)                        \
+  LINE(RISCV_Q, q, "q", RISCV_HWCAP_Q, 0)
+#define INTROSPECTION_PREFIX Riscv
+#define INTROSPECTION_ENUM_PREFIX RISCV
+#include "define_introspection_and_hwcaps.inl"
+
+////////////////////////////////////////////////////////////////////////////////
+// Implementation.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "internal/stack_line_reader.h"
+#include "internal/filesystem.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+
+static const RiscvInfo kEmptyRiscvInfo;
+
+static bool HandleRiscVLine(const LineResult result,
+                            RiscvInfo* const info) {
+  StringView line = result.line;
+  StringView key, value;
+  if (CpuFeatures_StringView_GetAttributeKeyValue(line, &key, &value)) {
+    if (CpuFeatures_StringView_IsEquals(key, str("isa"))) {
+      StringView tmp_sv;
+      value.ptr+=2; // ignore 'rv' prefix at the beginning
+      value.size-=2;
+      for (size_t i = 0; i < RISCV_LAST_; ++i){
+        tmp_sv.ptr = kCpuInfoFlags[i];
+        tmp_sv.size = strlen(kCpuInfoFlags[i]);
```

Comment:
Why are you using libc functions like `strlen`, `strcpy`? According to documentation `cpu_features is suitable for implementing fundamental libc functions like malloc, memcpy, and memcmp`. Hence we use our implementation, see `src/copy.inl`, `src/equals.inl`.

Reply:
@gchatelet, it seems I misunderstood. Is this the right point? Since I found several codes where `memmove` and `strlen` are used:
https://github.com/google/cpu_features/blob/main/src/stack_line_reader.c#L59 
https://github.com/google/cpu_features/blob/main/include/internal/string_view.h#L47

Reply:
It used to be strongly enforced but it is not anymore as it is hard to prevent the compiler from generating calls to these functions. That is, even if the code contains no call to `C` memory functions, the compiler may decide to generate them (e.g., https://godbolt.org/z/h5KKM7MEj)

Reply:
should we provide info to REAMDE that even with compiler flags such as `freestanding` `no-stdlib` `no-builtin` there is a possibility that the compiler will generate C functions?

Reply:
Yep I think so, here is an example where the compiler generates a call even with all the compiler flags https://godbolt.org/z/rTTfzEq15

In fact, compilers [require some C functions to exist](https://gcc.gnu.org/onlinedocs/gcc/Standards.html#:~:text=GCC%20requires%20the%20freestanding%20environment%20provide%20memcpy%2C%20memmove%2C%20memset%20and%20memcmp.).